images/spark/conf/entrypoint.sh

```diff
@@ -0,0 +1,159 @@
+# limitations under the License.
+#
+
+# echo commands to the terminal output
+set -ex
+
+# Check whether there is a passwd entry for the container UID
+myuid=$(id -u)
+mygid=$(id -g)
+# turn off -e for getent because it will return error code in anonymous uid case
+set +e
+uidentry=$(getent passwd $myuid)
+set -e
+
+# If there is no passwd entry for the container UID, attempt to create one
+if [ -z "$uidentry" ] ; then
+    if [ -w /etc/passwd ] ; then
+        echo "$myuid:x:$myuid:$mygid:anonymous uid:$SPARK_HOME:/bin/false" >> /etc/passwd
+    else
+        echo "Container ENTRYPOINT failed to add passwd entry for anonymous UID"
+    fi
+fi
+
+SPARK_K8S_CMD="$1"
+case "$SPARK_K8S_CMD" in
+    driver | driver-py | driver-r | executor)
+      shift 1
+      ;;
+    "")
+      ;;
+    *)
+      echo "Non-spark-on-k8s command provided, proceeding in pass-through mode..."
+      exec /sbin/tini -s -- "$@"
+      ;;
+esac
+
+SPARK_CLASSPATH="$SPARK_CLASSPATH:${SPARK_HOME}/jars/*"
+env | grep SPARK_JAVA_OPT_ | sort -t_ -k4 -n | sed 's/[^=]*=\(.*\)/\1/g' > /tmp/java_opts.txt
+readarray -t SPARK_EXECUTOR_JAVA_OPTS < /tmp/java_opts.txt
+
+if [ -n "$SPARK_EXTRA_CLASSPATH" ]; then
+  SPARK_CLASSPATH="$SPARK_CLASSPATH:$SPARK_EXTRA_CLASSPATH"
+fi
+
+if [ -n "$PYSPARK_FILES" ]; then
+    PYTHONPATH="$PYTHONPATH:$PYSPARK_FILES"
+fi
+
+PYSPARK_ARGS=""
+if [ -n "$PYSPARK_APP_ARGS" ]; then
+    PYSPARK_ARGS="$PYSPARK_APP_ARGS"
+fi
+
+R_ARGS=""
+if [ -n "$R_APP_ARGS" ]; then
+    R_ARGS="$R_APP_ARGS"
+fi
+
+if [ "$PYSPARK_MAJOR_PYTHON_VERSION" == "2" ]; then
+    pyv="$(python -V 2>&1)"
+    export PYTHON_VERSION="${pyv:7}"
+    export PYSPARK_PYTHON="python"
+    export PYSPARK_DRIVER_PYTHON="python"
+elif [ "$PYSPARK_MAJOR_PYTHON_VERSION" == "3" ]; then
+    pyv3="$(python3 -V 2>&1)"
+    export PYTHON_VERSION="${pyv3:7}"
+    export PYSPARK_PYTHON="python3"
+    export PYSPARK_DRIVER_PYTHON="python3"
+fi
+
+spark_submit_conf_params=(--conf "spark.driver.bindAddress=$SPARK_DRIVER_BIND_ADDRESS")
+
+function add_spark_conf_if_non_empty() {
+	if [[ -n "$2" ]]; then
+		spark_submit_conf_params+=(--conf "$1=$2")
+	fi
+}
+
+# This method is required for passing secret data to spark-submit via environment variables,
+# injected by K8s via Secrets.
+function configure_spark_properties() {
+    add_spark_conf_if_non_empty "spark.authenticate.secret" "${SPARK_AUTHENTICATE_SECRET}"
+    add_spark_conf_if_non_empty "spark.ssl.keyPassword" "${SPARK_SSL_KEYPASSWORD}"
+    add_spark_conf_if_non_empty "spark.ssl.keyStorePassword" "${SPARK_SSL_KEYSTOREPASSWORD}"
+    add_spark_conf_if_non_empty "spark.ssl.trustStorePassword" "${SPARK_SSL_TRUSTSTOREPASSWORD}"
+}
+
+configure_spark_properties
+
+# if SPARK_AUTHENTICATE_SECRET is set, enable RPC authetication for executors
+# and provide the auth secret via _SPARK_AUTH_SECRET variable
+if [[ -n "${SPARK_AUTHENTICATE_SECRET}" ]]; then
+  SPARK_EXECUTOR_JAVA_OPTS+=("-Dspark.authenticate=true")
```

Comment:
TBH, I'm not sure we should automate this for the users (because essentially we're hiding it by making it implicit) but not a blocker anyway

Reply:
I agree, but I think this is more like an implementation detail. If I, as a user, specify `spark.authenticate: true` in `sparkConf` for my application, I would expect the authentication will be enabled for driver and executors automatically, without any explicit configuration for the executor, except for the auth secret (and it is aligned with [the documentation](https://spark.apache.org/docs/latest/security.html#authentication)).